src/Struct/FindologicProduct.php

```diff
@@ -356,11 +356,17 @@ public function getKeywords(): array
 
     protected function setKeywords(): void
     {
-        $tags = $this->product->getTags();
-        if ($tags !== null && $tags->count() > 0) {
-            foreach ($tags as $tag) {
-                if (!Utils::isEmpty($tag->getName())) {
-                    $this->keywords[] = new Keyword($tag->getName());
+        $ignoreKeywords = [$this->product->getProductNumber(), $this->product->get('id'), $this->product
+        ->getManufacturer()->getName()];
+        $keywords = $this->product->getSearchKeywords();
+
+        if ($keywords !== null && $keywords->count() > 0) {
+            foreach ($keywords as $keyword) {
+                if (!Utils::isEmpty($keyword->getKeyword())) {
+                    $ignoreKeyword = in_array($keyword->getKeyword(), $ignoreKeywords);
```

Comment:
Instead of doing `$keyword->getKeyword()` in both lines, it is efficient to use variable instead.